packages/vuetify/src/components/VDatePicker/VDatePickerDateTable.ts

```diff
@@ -99,7 +94,7 @@ export default mixins(
         if (rows.length % (this.showWeek ? 8 : 7) === 0) {
           children.push(this.genTR(rows))
           rows = []
-          day < daysInMonth && this.showWeek && rows.push(this.genWeekNumber(weekNumber++))
+          day < daysInMonth && this.showWeek && rows.push(this.genWeekNumber(this.getWeekNumber(day + 7)))
```

Comment:
Do not use &&

Reply:
Rewritten to use if statement. && was used in the original implementation that's why I left it